install/kubernetes/helm/istio/charts/gateways/values.yaml

```diff
@@ -185,6 +185,8 @@ istio-mcgateway:
     name: tls
   envVariables:
     # this is needed to tell pilot to not generate
-    # mTLS clusters for the mcgateway
-    ISTIO_META_ISTIO_GATEWAY_MODE: multicluster
-
+    # mTLS clusters for the mcgateway except for
+    # istio-telemetry,istio-policy
+    # TODO: the FQDNs are hardcoded here. We need to find a way to
+    # parameterize the namespace and others.
+    ISTIO_META_ISTIO_MTLS_INCLUSION_ONLY: "istio-telemetry.istio-system.svc.cluster.local,istio-policy.istio-system.svc.cluster.local"
```

Comment:
I believe use of the service name (rather than FQDN) should be sufficient. Also these services are istio services. They can be an input argument to pilot if needed (to deal with additional services that may be added later). So an inclusion list doesn't seem to be necessary.

Although the comment indicates that the flag is specifically defined for mcgateway, the name itself ISTIO_META_ISTIO_MTLS_INCLUSION_ONLY is too generic. From the code change, it seems to be applicable to every proxy.

I still think that the flag ISTIO_META_ISTIO_GATEWAY_MODE is a better choice from both definition and implementation.

Reply:
The networking core code does not deal with short name services. It does not have the concept of namespaces either. So, FQDNs are the way to go. 

The Gateway mode will cause mtls to be dropped from all clusters. We still need another variable to tell us the services for which mTLs should not be dropped. 

So, this variable is in addition to the gateway mode if needed. I do not want to use this variable as well, as it opens the door for arbitrary customizations in the networking code based on ad hoc environment variables defined in each proxy (code becomes very hard to maintain over time).